spring-web/src/main/java/org/springframework/web/util/UrlPathHelper.java

```diff
@@ -392,12 +410,16 @@ else if (index1 == requestUri.length()) {
 	}
 
 	/**
-	 * Sanitize the given path. Uses the following rules:
+	 * Sanitize the given path if {code shouldSanitizePath()} is true.
+	 * Uses the following rules:
 	 * <ul>
 	 * <li>replace all "//" by "/"</li>
 	 * </ul>
 	 */
-	private static String getSanitizedPath(final String path) {
+	private String getSanitizedPath(final String path) {
```

Comment:
This was a **private** static method that had 2 usages in the class, made it non-static to use the flag.
I can abstract the logic into a static method if you think that makes sense.